client/client_test.go

```diff
@@ -1370,64 +1351,117 @@ func testPublishNoOneHasSnapshotKey(t *testing.T, rootType string) {
 
 	// Publish something
 	addTarget(t, repo, "v1", "../fixtures/intermediate-ca.crt")
-	err = repo.Publish()
+	err := repo.Publish()
 	assert.Error(t, err)
 	assert.IsType(t, validation.ErrBadHierarchy{}, err)
 }
 
-// If the snapshot metadata is corrupt, whether the client or server has the
-// snapshot key, we can't publish.
-// We test this with both an RSA and ECDSA root key
+// If the snapshot metadata is corrupt or the snapshot metadata is unreadable,
+// we can't publish for the first time (whether the client or server has the
+// snapshot key), because there is no existing data for us to download. If the
+// repo has already been published, it doesn't matter if the metadata is corrupt
+// because we can just redownload if it is.
 func TestPublishSnapshotCorrupt(t *testing.T) {
-	testPublishBadExistingSnapshot(t, data.ECDSAKey, true, true)
-	testPublishBadExistingSnapshot(t, data.ECDSAKey, false, true)
-}
+	ts := fullTestServer(t)
+	defer ts.Close()
 
-// If the snapshot metadata is unreadable, whether the client or server has the
-// snapshot key, we can't publish.
-// We test this with both an RSA and ECDSA root key
-func TestPublishSnapshotUnreadable(t *testing.T) {
-	testPublishBadExistingSnapshot(t, data.ECDSAKey, true, false)
-	testPublishBadExistingSnapshot(t, data.ECDSAKey, false, false)
+	// do not publish first - publish should fail
+	repo, _ := initializeRepo(t, data.ECDSAKey, "docker.com/notary1", ts.URL, true)
+	defer os.RemoveAll(repo.baseDir)
+	testPublishBadMetadata(t, data.CanonicalSnapshotRole, repo, false, false)
```

Comment:
The difference between 1 and 2 is server snapshot signing vs not, similarly with 3 and 4.  I'll add comments.